File: seeyousun-backend/resort-service/src/main/java/com/tass/seeyousun/resortservice/model/Event.java
package com.tass.seeyousun.resortservice.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "events")
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @Temporal(TemporalType.DATE)
    private LocalDate initialDate;

    @Temporal(TemporalType.DATE)
    private LocalDate finalDate;

    private String description;

    @ManyToOne
    @JoinColumn(name = "resort_id")
    @JsonBackReference
    private Resort resort;
}

File: frontend-app/src/app/models/event.ts
export class Event {
    id: number;
    name: string;
    initialDateTime: Date;
    finalDate: Date;
    description: string;
    duration: number;
    isFull: boolean;

    constructor(id: number, name: string, initialDateTime: Date, finalDate: Date, description: string, duration: number, isFull: boolean) {
        this.id = id;
        this.name = name;
        this.initialDateTime = initialDateTime;
        this.finalDate = finalDate;
        this.description = description;
        this.duration = duration;
        this.isFull = isFull;
    }
}